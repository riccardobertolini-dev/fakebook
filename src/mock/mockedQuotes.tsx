export const MockedQuote1 = {
	text: 'Genius is one percent inspiration and ninety-nine percent perspiration.',
	author: 'Thomas Edison',
};
export const MockedQuote2 = {
	text: 'You can observe a lot just by watching.',
	author: 'Yogi Berra',
};
