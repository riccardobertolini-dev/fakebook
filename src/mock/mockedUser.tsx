export const MockedUser1 = {
	gender: 'female',
	name: {
		title: 'Ms',
		first: 'Isabelle',
		last: 'Davies',
	},
	location: {
		street: {
			number: 8255,
			name: 'Esk Street',
		},
		city: 'Taupo',
		state: 'West Coast',
		country: 'New Zealand',
		postcode: 25207,
		coordinates: {
			latitude: '53.3834',
			longitude: '16.3516',
		},
		timezone: {
			offset: '-6:00',
			description: 'Central Time (US & Canada), Mexico City',
		},
	},
	email: 'isabelle.davies@example.com',
	login: {
		uuid: '71d3d4bf-2c47-450f-b3bf-e15a7fef23da',
		username: 'yellowtiger203',
		password: 'pyramid',
		salt: 'WrEua7np',
		md5: '8d185b252bdc6739cc36a7d4c67bcb1f',
		sha1: '4843e0607dd75b952504e278013508847f002237',
		sha256: 'dde6119b6dad7740579c5ace7f89789b7c24973dd203285bac2028018a046926',
	},
	dob: {
		date: '1965-10-08T18:06:10.878Z',
		age: 57,
	},
	registered: {
		date: '2015-12-03T15:58:10.458Z',
		age: 6,
	},
	phone: '(999)-128-6616',
	cell: '(132)-372-7000',
	id: {
		name: '',
		value: null,
	},
	picture: {
		large: 'https://randomuser.me/api/portraits/women/44.jpg',
		medium: 'https://randomuser.me/api/portraits/med/women/44.jpg',
		thumbnail: 'https://randomuser.me/api/portraits/thumb/women/44.jpg',
	},
	nat: 'NZ',
};
