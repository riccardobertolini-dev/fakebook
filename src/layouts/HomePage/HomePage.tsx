import React, { useState } from 'react';
import { User } from '../../models/User';
import { getUsers } from '../../api/getUsers';
import { Body, SocialWall, ShowMoreBtn, Footer } from '../../App.styles';
import { UserTile } from '../../components/UserTile/UserTile';
import { useGlobalContext } from '../../globalContext';
import { UserTileLoading } from '../../components/UserTile/UserTileLoading';
import { UserInteractiveTile } from '../../components/UserInteractiveTile/UserInteractiveTile';
import { MockedUser2 } from '../../mock/mockedUser';

const HomePage = () => {
	const [loading, setLoading] = useState<boolean>(false);
	const [error, setError] = useState<boolean>(false);
	const [page, setPage] = useState<number>(1);
	const [userList, updateUserList] = useState<User[]>([]);
	const { users, updateUsers, userPosts } = useGlobalContext();

	React.useEffect(() => {
		const fetchUsers = async () => {
			if (page === 1 && users.length !== 0) {
				updateUserList(users);
			} else {
				setLoading(true);
				const fetchedUsers = await getUsers();
				if (typeof fetchedUsers !== 'string') {
					const updatedList = [...userList, ...fetchedUsers.results];
					updateUserList(updatedList);
					updateUsers(updatedList);
				} else {
					setError(true);
				}
				setLoading(false);
			}
		};

		fetchUsers();
	}, [page]);

	if (error) return <div>error, sorry!</div>;

	return (
		<Body>
			<UserInteractiveTile />
			<SocialWall>
				{loading && <UserTileLoading times={5} />}
				{[...userPosts, ...userList]?.map((user, index) => {
					if (typeof user === 'string') {
						return <UserTile user={MockedUser2} index={index} hardCodedquote={user} selfContent={true} />;
					}
					return <UserTile user={user} index={index} selfContent={false} />;
				})}
				<Footer>
					<ShowMoreBtn onClick={() => setPage(page + 1)}>Show More</ShowMoreBtn>
				</Footer>
			</SocialWall>
		</Body>
	);
};

export default HomePage;
