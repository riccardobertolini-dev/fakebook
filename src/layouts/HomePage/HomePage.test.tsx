import * as React from 'react';
import renderer from 'react-test-renderer';
import HomePage from './HomePage';
import { MockedUser1 } from '../../mock/mockedUser';

jest.mock('../../api/getUsers', () => ({
	getUsers: jest.fn().mockReturnValue(() => [MockedUser1]),
}));

jest.mock('axios');

describe('HomePage component', () => {
	it('should renders', () => {
		const wrapper = renderer.create(<HomePage />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
