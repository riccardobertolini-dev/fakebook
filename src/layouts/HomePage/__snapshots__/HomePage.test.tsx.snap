// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should renders 1`] = `
<div
  className="sc-gswNZR kDQAfP"
>
  <div
    className="sc-bBABsx kEvVia"
  >
    <div
      className="sc-cwSeag hGIzYU"
    >
      <img
        alt=""
        src="/generic-user.png"
      />
    </div>
    <input
      className="sc-iveFHk kwvCaR"
      onChange={[Function]}
      placeholder="What's in your Mind, John?"
      type="text"
      value=""
    />
    <button
      className="sc-lllmON iuLelm"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
        />
      </svg>
    </button>
  </div>
  <div
    className="sc-dkrFOg lkTwKs"
  >
    <div
      className="sc-kDvujY fwoiUJ"
    >
      <button
        className="sc-jrcTuL drZSHS"
        onClick={[Function]}
      >
        Show More
      </button>
    </div>
  </div>
</div>
`;
